Tests/LowLevel Test/TestGR32_LowLevel.pas

```diff
@@ -1341,6 +1341,9 @@ procedure TTestMath.TestSign;
 var
   Index : Integer;
 begin
+  CheckEquals(0, GR32_Math.Sign(0));
+  CheckEquals(1, GR32_Math.Sign($7FFFFFFF));
+  CheckEquals(-1, GR32_Math.Sign(Integer($80000000)));
```

Comment:
The PUREPASCAL version fails for the test case `Sign($80000000) = -1`.
However, that is an existing problem which your added tests has surfaced, so just FYI.

Reply:
Does Delphi have an equivalent to Free Pascal's "SarLongint" intrinsic?  That would permit the use of arithmetic shifts.  If not, it might have to be an explicit check for $80000000, since negating that value causes an overflow.

Reply:
Not really. There is only `x div 2^n` which is compiled to SAR on 64-bit but unfortunately:
https://github.com/graphics32/graphics32/blob/8265594faf795163da9ea32a0bcd559605876c35/Source/GR32_LowLevel.pas#L252-L267

I think that if we have to test for a specific value then that will outweigh the benefit of the trick applied. Better to just document it.

Reply:
I'll have a think on that one.  The Sign function behaving differently between assembly and Pascal implementations, even if just for one extreme value, is not ideal.

Reply:
We could just defer to the RTL Sign function and let inlining sort the overhead out.
That is already done in a few other places where a faster PUREPASCAL version isn't possible.

Reply:
>  behaving differently between assembly and Pascal implementations, even if just for one extreme value, is not ideal.

You are going to really enjoy the blend/merge functions then :-)

Reply:
> We could just defer to the RTL Sign function and let inlining sort the overhead out. That is already done in a few other places where a faster PUREPASCAL version isn't possible.

I've pushed an extra commit that defers to Math.Sign under PUREPASCAL.  Does this work for you?